File: src/java/eu/sig/training/ch05/buildandsendmail/v1/BuildAndSendMail.java
package eu.sig.training.ch05.buildandsendmail.v1;

public class BuildAndSendMail {
    // tag::buildAndSendMail[]
    public void buildAndSendMail(MailMan m, String firstName, String lastName,
        String division, String subject, MailFont font, String message1,
        String message2, String message3) {
        // Format the email address
        String mId = firstName.charAt(0) + "." + lastName.substring(0, 7) + "@"
            + division.substring(0, 5) + ".compa.ny";
        // Format the message given the content type and raw message
        MailMessage mMessage = formatMessage(font,
            message1 + message2 + message3);
        // Send message
        m.send(mId, subject, mMessage);
    }
    // end::buildAndSendMail[]

    @SuppressWarnings("unused")
    private MailMessage formatMessage(MailFont font, String string) {
        return null;
    }

    private class MailMan {

        @SuppressWarnings("unused")
        public void send(String mId, String subject, MailMessage mMessage) {}

    }

    private class MailFont {

    }

    private class MailMessage {

    }

}

File: src/csharp/eu/sig/training/ch05/buildandsendmail/v1/BuildAndSendMail.cs
namespace eu.sig.training.ch05.buildandsendmail.v1
{
    public class BuildAndSendMail
    {
        // tag::buildAndSendMail[]
        public void DoBuildAndSendMail(MailMan m, string firstName, string lastName,
            string division, string subject, MailFont font, string message1,
            string message2, string message3)
        {
            // Format the email address
            string mId = $"{firstName[0]}.{lastName.Substring(0, 7)}" +
                $"@{division.Substring(0, 5)}.compa.ny";
            // Format the message given the content type and raw message
            MailMessage mMessage = FormatMessage(font,
                message1 + message2 + message3);
            // Send message
            m.Send(mId, subject, mMessage);
        }
        // end::buildAndSendMail[]

        public MailMessage FormatMessage(MailFont font, string s)
        {
            return null;
        }

        public class MailMan
        {

            public void Send(string mId, string subject, MailMessage mMessage) { }

        }

        public class MailFont
        {

        }

        public class MailMessage
        {

        }

    }
}
